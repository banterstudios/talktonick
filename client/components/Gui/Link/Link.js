import React, { Component } from 'react'
import glamorous from 'glamorous'
import PropTypes from 'prop-types'
import { Link as routerLink } from 'react-router-dom'

const getStyles = ({ theme: { ctaPrimary, ctaHighlight, fontSizeText, fontPrimary, fontSizeMdText }, isActive, modifier }) => ({
  position: 'relative',
  color: (isActive ? ctaHighlight : ctaPrimary),
  fontSize: (modifier === 'sm') ? fontSizeMdText : fontSizeText,
  fontFamily: fontPrimary,
  textDecoration: 'none',
  transition: 'color .4s ease-out',
  ':hover': {
    color: ctaHighlight
  }
})

const StyledLink = glamorous.a(getStyles)
const StyledRouterLink = glamorous(routerLink)(getStyles)

class Link extends Component {
  static propTypes = {
    href: PropTypes.string,
    linkType: PropTypes.string,
    target: PropTypes.string,
    children: PropTypes.any,
    isActive: PropTypes.bool
  }

  static defaultProps = {
    href: '/',
    target: '_blank',
    linkType: 'route'
  }

  renderNativeLink = ({ children, href, target, isActive, modifier }) => (
    <StyledLink href={href} target={target} isActive={isActive} modifier={modifier}>
      { children }
    </StyledLink>
  )

  renderRouterLink = ({ href, children, isActive, modifier }) => (
    <StyledRouterLink to={href} isActive={isActive} modifier={modifier}>
      { children }
    </StyledRouterLink>
  )

  render () {
    const { linkType, ...rest } = this.props

    const renderLookup = {
      native: this.renderNativeLink,
      route: this.renderRouterLink
    }

    return renderLookup[linkType] ? renderLookup[linkType](rest) : null
  }
}

export default Link
