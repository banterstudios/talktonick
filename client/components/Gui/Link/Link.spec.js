import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Link from './Link'
import { MemoryRouter as Router } from 'react-router-dom'

describe('components/Gui/Link', () => {
  describe('@renders', () => {
    it('should render correctly', () => {
      const component = shallow(<Router><Link /></Router>)

      expect(toJson(component.dive())).toMatchSnapshot()
      component.unmount()
    })
  })
})
