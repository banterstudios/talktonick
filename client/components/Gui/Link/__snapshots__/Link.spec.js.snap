// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Gui/Link @renders should render correctly 1`] = `
<Link
  href="/"
  linkType="route"
  target="_blank"
/>
`;
